translate/cloud-client/snippets.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to perform basic operations with the
+Google Cloud Translate API
+
+For more information, the documentation at
+https://cloud.google.com/translate/docs.
+"""
+
+import argparse
+
+from google.cloud import translate
+
+
+def detect_language(api_key, text):
+    """Detects the text's language."""
+    translate_client = translate.Client(api_key)
+
+    # Text can also be a sequence of strings, in which case this method
+    # will return a sequence of results for each text.
+    result = translate_client.detect_language(text)
+
+    print('Text: {}'.format(text))
+    print('Confidence: {}'.format(result['confidence']))
+    print('Language: {}'.format(result['language']))
+
+
+def list_languages(api_key):
+    """Lists all available langauges."""
+    translate_client = translate.Client(api_key)
+
+    results = translate_client.get_languages()
+
+    for language in results:
+        print(u'{name} ({language})'.format(**language))
+
+
+def list_languages_with_target(api_key, target):
+    """Lists all available langauges and localizes them to the target language.
+
+    Target must be an ISO 639-1 language code.
```

Comment:
Might also provide a link to //g.co/cloud/translate/v2/translate-reference#supported_languages since I don't think _every_ language is supported.